### ISSUE TITLE
[iOS] waitForRender on showModal breaks modal layout and dismiss

### ISSUE THREAD

**Issue by FransTwisk** [wix/react-native-navigation#7162]
## 🐛 Bug Report
On iOS, using `waitForRender` on `showModal` results in broken modal rendering, and when dismissing a modal it results in a black screen on previous (root) stack.

### Have you read the [Contributing Guidelines on issues](https://wix.github.io/react-native-navigation/docs/meta-contributing/)?
Yes

## To Reproduce
1. Use following on (global) navigator options:
```
  animations: {
    showModal: {
      waitForRender: false,
    },
  },
```
2. Open modal in iOS
3. Dismiss modal in iOS

## Expected behavior
Using `waitForRender: true` on `showModal` should not affect the (default) layout of modals on iOS, and should just return to previous stack, instead or rendering a black screen when dismissing.

## Actual Behavior
1. Modal is rendered sticked to top instead of bottom of screen.
<img width="551" alt="Screenshot 2021-06-21 at 20 51 59" src="https://user-images.githubusercontent.com/76724/122813236-004e5c80-d2d3-11eb-8084-9e59344b15d4.png">

2. When dismissing, the previous stack is not rendered, but black.
<img width="551" alt="Screenshot 2021-06-21 at 20 52 02" src="https://user-images.githubusercontent.com/76724/122813259-07756a80-d2d3-11eb-87f6-5c901cadd3e1.png">

## Your Environment
- React Native Navigation version: 7.16.0
- React Native version: 0.64.2
- Platform(s) (iOS, Android, or both?): iOS
- Device info (Simulator/Device? OS version? Debug/Release?): Simulator & Device, iOS 12 (probably others as well), Debug & Release

**Comment by yogevbd** [comment]
@FransTwisk Could you please reproduce it in the playground app? I tried with no luck

**Comment by tomhicks** [comment]
I just replicated exactly this behaviour.

```javascript
      animations: {
        push: {
          waitForRender: true,
        },
        // showModal: {
        //   waitForRender: true,
        // },
      },
```

The comments fix everything!

**Comment by wsliaw** [comment]
this issue fixed in 7.17.1, but waitForRender seems broken
